### ISSUE TITLE
Signed public keys not working

### ISSUE THREAD

**Issue by toddrosner** [saltstack/salt#31600]
### Description of Issue/Question

Multi-Master PKI not working

### Setup
Not able to get signed public keys to work as mentioned in the link below. When configured, the Minion will always be put into the `Unaccepted` state.
https://docs.saltstack.com/en/latest/topics/tutorials/multimaster_pki.html

### Steps to Reproduce Issue
Follow this link https://docs.saltstack.com/en/latest/topics/tutorials/multimaster_pki.html

The command salt-minion -l debug produces the following output:
```
[root@ip-10-0-1-120 minion]# salt-minion -l debug
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: ip-10-0-1-120.us-west-2.compute.internal
[DEBUG   ] Configuration file path: /etc/salt/minion
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[INFO    ] Setting up the Salt Minion "ip-10-0-1-120.us-west-2.compute.internal"
[DEBUG   ] Created pidfile: /var/run/salt-minion.pid
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] The `dmidecode` binary is not available on the system. GPU grains will not be available.
[DEBUG   ] Attempting to authenticate with the Salt Master at 10.0.1.236
[DEBUG   ] Initializing new SAuth for ('/etc/salt/pki/minion', 'ip-10-0-1-120.us-west-2.compute.internal', 'tcp://10.0.1.236:4506')
[INFO    ] Generating keys: /etc/salt/pki/minion
[ERROR   ] The Salt Master has cached the public key for this node, this salt minion will wait for 10 seconds before attempting to re-authenticate
[INFO    ] Waiting 10 seconds before retry.
[ERROR   ] The Salt Master has cached the public key for this node, this salt minion will wait for 10 seconds before attempting to re-authenticate
[INFO    ] Waiting 10 seconds before retry.
```

### Versions Report
```
[root@ip-10-0-1-236 master]# salt --versions-report
           Salt: 2015.5.9
         Python: 2.6.9 (unknown, Dec 17 2015, 01:08:55)
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 3.2.5
           Mako: Not Installed
        Tornado: Not Installed
        timelib: Not Installed
       dateutil: Not Installed
```
Any advice is much appreciated.

**Comment by jfindlay** [comment]
@toddrosner, thanks for reporting.  This looks like a bug to me.

**Comment by cachedout** [comment]
I don't see why this is a bug. A key is going to be unaccepted until it's explicitly accepted (or the master is running with a configuration to accept any key, such as open mode). Simply having the minion verify the public key of the master doesn't imply that the master should trust any key that it receives from the minion. After accepting the key on the master using `salt-key`, does it work as intended?

**Comment by toddrosner** [comment]
@cachedout Yes everything works as intended by accepting the key using salt-key. Everything also works as intended if the master is configured to auto_accept keys...have not tested open mode though.

Maybe I'm misunderstanding the design. I had assumed that by having the master_sign.pub key on the minions, that there would not need to be any other setting required for minions to be accepted. 

Is it by design that the combination of the following two configs facilitates open and secured minion acceptance, provided that they master_sign.pub key is on the minion?

master config:
`auto_accept: True`

minion config:
`verify_master_pubkey_sign: True`

**Comment by cachedout** [comment]
Having the master key signature on the minions is to verify the authenticity of the master. However, the master still needs to verify the authenticity and accept the key from the minion. To skip the step of manually accepting the key on the master side, you'd need to either run in open mode, auto accept, or pre-seed minion keys on the master.

**Comment by toddrosner** [comment]
Thanks for confirming this @cachedout.

Pre-seeding won't work, as all instances are in auto scaling groups. Open mode sounds like it's a bit too wide-open in allowing all authentication forms. Auto accept is what I originally had in-place, and this works well in a secured VPC where all minions accessing the master are actually supposed to.

It would be great if a minion could get a master-minion key or token from secrets management and securely authenticate to the master (without auto accept). This way minions could verify the master, and masters could verify the minions, and all of this could happen securely with auto scaling.
